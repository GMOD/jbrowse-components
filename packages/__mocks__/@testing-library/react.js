import { StylesProvider } from '@material-ui/core/styles'
import React from 'react'

const react = jest.requireActual('@testing-library/react')
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`
const render = args => {
  return react.render(
    <StylesProvider generateClassName={generateClassName}>
      {args}
    </StylesProvider>,
  )
}

module.exports = { ...react, render }
