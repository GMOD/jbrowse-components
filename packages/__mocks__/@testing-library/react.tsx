import React, { Suspense } from 'react'
import { createTheme, ThemeProvider } from '@mui/material/styles'

const react = jest.requireActual('@testing-library/react')
const render = (args: React.ReactNode) => {
  return react.render(
    <Suspense fallback={null}>
      <ThemeProvider theme={createTheme()}>{args}</ThemeProvider>
    </Suspense>,
  )
}

module.exports = { ...react, render }
