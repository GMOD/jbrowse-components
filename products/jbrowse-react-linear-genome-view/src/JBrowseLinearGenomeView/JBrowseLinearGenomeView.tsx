import { observer } from 'mobx-react'
import { Instance } from 'mobx-state-tree'
import React from 'react'
import createSessionModel from '../createModel/createSessionModel'
import ModalWidget from './ModalWidget'
import ViewContainer from './ViewContainer'

type Session = Instance<ReturnType<typeof createSessionModel>>

function JBrowseLinearGenomeView({
  viewState,
}: {
  viewState: { session: Session }
}) {
  const { session } = viewState
  const { view, pluginManager } = session
  const viewType = pluginManager.getViewType(view.type)
  if (!viewType) {
    throw new Error(`unknown view type ${view.type}`)
  }
  const { ReactComponent } = viewType

  return (
    <div>
      <ViewContainer key={`view-${view.id}`} view={view}>
        <ReactComponent model={view} session={session} />
      </ViewContainer>
      <ModalWidget session={session} />
    </div>
  )
}

export default observer(JBrowseLinearGenomeView)
