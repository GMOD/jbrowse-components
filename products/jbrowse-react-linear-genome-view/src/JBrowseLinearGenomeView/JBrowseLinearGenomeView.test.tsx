import { render } from '@testing-library/react'
import React from 'react'
import { createViewState } from '..'
import JBrowseLinearGenomeView from './JBrowseLinearGenomeView'

window.requestIdleCallback = (cb: (deadline: IdleDeadline) => void) => {
  cb({ didTimeout: false, timeRemaining: () => 0 })
  return 0
}
window.cancelIdleCallback = () => {}
window.requestAnimationFrame = cb => setTimeout(cb)
window.cancelAnimationFrame = () => {}

const assembly = {
  name: 'volvox',
  sequence: {
    type: 'ReferenceSequenceTrack',
    trackId: 'volvox_refseq',
    adapter: {
      type: 'FromConfigSequenceAdapter',
      features: [
        {
          refName: 'ctgA',
          uniqueId: 'firstId',
          start: 0,
          end: 120,
          seq:
            'cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgattatgattggttctttagccttggtttagattggtagtagtagcggcgctaatgctacctg',
        },
      ],
    },
  },
}

const defaultSession = {
  name: 'Test',
  view: {
    id: 'test_view',
    type: 'LinearGenomeView',
    bpPerPx: 0.05,
    displayedRegions: [
      {
        refName: 'ctgA',
        start: 0,
        end: 120,
        reversed: false,
        assemblyName: 'volvox',
      },
    ],
    tracks: [
      {
        type: 'ReferenceSequenceTrack',
        configuration: 'volvox_refseq',
        displays: [
          {
            type: 'LinearReferenceSequenceDisplay',
            configuration: 'volvox_refseq-LinearReferenceSequenceDisplay',
          },
        ],
      },
    ],
  },
}

describe('<JBrowseLinearGenomeView />', () => {
  it('renders successfully', async () => {
    const state = createViewState({
      assembly,
      tracks: [],
      defaultSession,
    })
    state.session.view.setWidth(800)
    const { container, findByText } = render(
      <JBrowseLinearGenomeView viewState={state} />,
    )
    await findByText(/Reference Sequence/)
    expect(container.firstChild).toMatchSnapshot()
  })
})
