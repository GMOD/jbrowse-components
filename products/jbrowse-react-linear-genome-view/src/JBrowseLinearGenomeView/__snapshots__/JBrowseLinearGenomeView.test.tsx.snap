// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JBrowseLinearGenomeView /> renders successfully 1`] = `
<div
  class="makeStyles-avoidParentStyle"
>
  <div
    class="MuiScopedCssBaseline-root"
  >
    <div
      class="MuiPaper-root makeStyles-viewContainer MuiPaper-elevation12 MuiPaper-rounded"
      style="padding: 0px 4px 4px;"
    >
      <div
        style="display: flex;"
      >
        <button
          aria-controls="view-menu"
          aria-haspopup="true"
          aria-label="more"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-iconRoot MuiIconButton-sizeSmall MuiIconButton-edgeStart"
          data-testid="view_menu_icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="makeStyles-grow"
        />
        <div
          class="makeStyles-grow"
        />
        <div
          style="width: 20px; height: 20px;"
        >
          <svg
            viewBox="0 0 175 175"
          >
            <path
              d="M120.5,57.5c.168,0,.333.012.5.013V40.506c-.167,0-.333-.006-.5-.006a98.325,98.325,0,0,0-77.4,37.565q-2.664,3.378-5.023,6.994l14.206,9.327Q54.043,91.709,56,89.179A81.363,81.363,0,0,1,120.5,57.5Z"
              style="fill: #ffffff;"
            />
            <path
              d="M148.668,86.223c.308.133.6.283.908.42l6.746-15.618c-.153-.067-.3-.143-.454-.21a81.36,81.36,0,0,0-71.777,3.5q-2.8,1.544-5.469,3.309l9.249,14.054A63.93,63.93,0,0,1,148.668,86.223Z"
              style="fill: #e6e6e6;"
            />
            <path
              d="M170.894,109.37A63.929,63.929,0,0,0,113.4,88.861l2.336,15.474a48.456,48.456,0,0,1,44.322,15.42c.111.125.214.255.324.381l11.16-10C171.323,109.879,171.118,109.62,170.894,109.37Z"
              style="fill: #cccccc;"
            />
            <path
              d="M150.921,150.314c.04.163.07.326.107.489l16.509-4.058c-.039-.163-.067-.326-.107-.489A48.457,48.457,0,0,0,135.575,111.8l-5.515,15.591A32.466,32.466,0,0,1,150.921,150.314Z"
              style="fill: #b3b3b3;"
            />
            <path
              d="M134.1,138.129l-9.779,13.213a32.486,32.486,0,0,1,7.823,14.463l15.147.852c.012-.166.032-.331.041-.5A32.464,32.464,0,0,0,134.1,138.129Z"
              style="fill: #999999;"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          style="position: relative;"
        >
          <div>
            <div
              style="position: relative;"
            >
              <div
                class="makeStyles-rubberBandControl"
                role="presentation"
              >
                <div
                  class="makeStyles-scaleBar"
                >
                  <div
                    class="makeStyles-scaleBarVisibleRegion"
                    style="width: 267px; left: 0px; background: rgba(66, 119, 127, 0.3); border-color: rgb(66, 119, 127);"
                  />
                  <div
                    class="makeStyles-scaleBarContig"
                    style="width: 0px; left: 0px; background-color: rgb(153, 153, 153);"
                  />
                  <div>
                    <p
                      class="MuiTypography-root makeStyles-scaleBarRefName MuiTypography-body1"
                      style="left: 3px; color: rgb(153, 102, 0);"
                    >
                      ctgA
                    </p>
                    <div
                      class="makeStyles-scaleBarContig makeStyles-scaleBarContigForward makeStyles-scaleBarBorder"
                      style="left: 0px; width: 800px; border-color: rgb(153, 102, 0);"
                    >
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 133.33333333333334px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        20
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 266.6666666666667px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        40
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 400px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        60
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 533.3333333333334px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        80
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 666.6666666666667px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        100
                      </p>
                      <p
                        class="MuiTypography-root makeStyles-scaleBarLabel MuiTypography-body2"
                        style="left: 800px; pointer-events: none; color: rgb(153, 102, 0);"
                      >
                        120
                      </p>
                    </div>
                  </div>
                  <div
                    class="makeStyles-scaleBarContig"
                    style="width: 0px; left: 800px; background-color: rgb(153, 153, 153);"
                  />
                </div>
              </div>
            </div>
            <div
              class="makeStyles-overview"
            >
              <svg
                class="makeStyles-overviewSvg"
                height="48"
              >
                <polygon
                  fill="rgba(66, 119, 127, 0.3)"
                  points="0,48,801,48,267,0,0,0"
                  stroke="rgba(66, 119, 127, 0.8)"
                />
              </svg>
              <div
                class="makeStyles-headerBar"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-toggleButton MuiButton-textSecondary MuiButton-textSizeSmall MuiButton-sizeSmall"
                  tabindex="0"
                  title="Open track selector"
                  type="button"
                  value="track_select"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-buttonSpacer MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M21 19v-2H8v2h13m0-6v-2H8v2h13M8 7h13V5H8v2M4 5v2h2V5H4M3 5a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H4a1 1 0 01-1-1V5m1 6v2h2v-2H4m-1 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H4a1 1 0 01-1-1v-2m1 6v2h2v-2H4m-1 0a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 01-1 1H4a1 1 0 01-1-1v-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="makeStyles-spacer"
                />
                <div
                  class="MuiFormGroup-root makeStyles-headerForm MuiFormGroup-row"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-panButton MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-panButton MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    aria-expanded="false"
                    class="MuiAutocomplete-root"
                    data-testid="autocomplete"
                    role="combobox"
                    style="width: 200px;"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root makeStyles-headerRefName MuiFormControl-marginDense MuiFormControl-fullWidth"
                      style="margin: 7px; min-width: 175px;"
                    >
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                        style="padding: 0px; height: 32px; background: rgba(255, 255, 255, 0.8);"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-invalid="false"
                          autocapitalize="none"
                          autocomplete="off"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                          id="refNameAutocomplete-test_view"
                          placeholder="Search for location"
                          spellcheck="false"
                          type="text"
                          value="ctgA:1..40"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                          style="margin-right: 7px;"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                            />
                          </svg>
                          <button
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <div
                          class="MuiAutocomplete-endAdornment"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend"
                            style="width: 0.01px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <p
                  class="MuiTypography-root makeStyles-bp MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  40 bp
                </p>
                <div
                  class="makeStyles-container"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                    data-testid="zoom_out"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14zM7 9h5v1H7z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <span
                    class="MuiSlider-root MuiSlider-colorPrimary makeStyles-slider"
                  >
                    <span
                      class="MuiSlider-rail"
                    />
                    <span
                      class="MuiSlider-track"
                      style="left: 0%; width: 56.78411120929245%;"
                    />
                    <input
                      type="hidden"
                      value="432.19280948873626"
                    />
                    <span
                      aria-orientation="horizontal"
                      aria-valuemax="564.3856189774724"
                      aria-valuemin="258.49625007211563"
                      aria-valuenow="432.19280948873626"
                      class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                      data-index="0"
                      role="slider"
                      style="left: 56.78411120929245%;"
                      tabindex="0"
                    />
                  </span>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                    data-testid="zoom_in"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                        <path
                          d="M12 10h-2v2H9v-2H7V9h2V7h1v2h2v1z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="makeStyles-spacer"
                />
              </div>
            </div>
          </div>
          <div
            class="makeStyles-tracksContainer"
            role="presentation"
          >
            <div
              class="makeStyles-verticalGuidesZoomContainer"
            >
              <div
                class="makeStyles-verticalGuidesContainer"
                style="left: -1024px; width: 2064px;"
              >
                <div
                  class="makeStyles-boundaryPaddingBlock"
                  style="width: 1024px;"
                />
                <div
                  class="makeStyles-contentBlock"
                  style="width: 1040px;"
                >
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: -20px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 0px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 20px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 40px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 60px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 80px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 100px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 120px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 140px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 160px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: 180px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 200px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 220px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 240px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 260px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 280px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 300px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 320px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 340px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 360px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: 380px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 400px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 420px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 440px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 460px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 480px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 500px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 520px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 540px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 560px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: 580px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 600px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 620px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 640px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 660px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 680px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 700px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 720px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 740px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 760px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: 780px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 800px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 820px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 840px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 860px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 880px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 900px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 920px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 940px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 960px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-majorTick"
                    style="left: 980px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 1000px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 1020px;"
                  />
                  <div
                    class="makeStyles-tick makeStyles-minorTick"
                    style="left: 1040px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="makeStyles-rubberBandControl"
              data-testid="rubberBand_controls"
              role="presentation"
            >
              <div
                class="MuiPaper-root makeStyles-scaleBarContainer MuiPaper-outlined MuiPaper-rounded"
                data-resizer="true"
                style="height: 17px; box-sizing: border-box;"
              >
                <div
                  class="makeStyles-scaleBarZoomContainer"
                >
                  <div
                    class="makeStyles-scaleBar"
                    style="left: -1025px; width: 2064px; height: 17px; box-sizing: border-box;"
                  >
                    <div
                      class="makeStyles-boundaryPaddingBlock"
                      style="background: none; width: 1024px;"
                    />
                    <div
                      class="makeStyles-contentBlock"
                      style="width: 1040px;"
                    >
                      <div
                        class="makeStyles-tick"
                        style="left: -20px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          0
                        </p>
                      </div>
                      <div
                        class="makeStyles-tick"
                        style="left: 180px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          10
                        </p>
                      </div>
                      <div
                        class="makeStyles-tick"
                        style="left: 380px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          20
                        </p>
                      </div>
                      <div
                        class="makeStyles-tick"
                        style="left: 580px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          30
                        </p>
                      </div>
                      <div
                        class="makeStyles-tick"
                        style="left: 780px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          40
                        </p>
                      </div>
                      <div
                        class="makeStyles-tick"
                        style="left: 980px;"
                      >
                        <p
                          class="MuiTypography-root makeStyles-majorTickLabel MuiTypography-body1"
                        >
                          50
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
                <p
                  class="MuiTypography-root makeStyles-refLabel MuiTypography-body1"
                  data-testid="refLabel-ctgA"
                  style="left: -1px; padding-left: 1px;"
                >
                  ctgA
                </p>
              </div>
            </div>
            <div
              class="makeStyles-spacer"
            />
            <div
              class="makeStyles-root"
            >
              <div
                class="MuiPaper-root makeStyles-trackLabel makeStyles-trackLabelOverlap makeStyles-root MuiPaper-elevation1 MuiPaper-rounded"
              >
                <span
                  class="makeStyles-dragHandle"
                  data-testid="dragHandle-test_view-volvox_refseq"
                  draggable="true"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-dragHandleIcon MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <button
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                  tabindex="0"
                  title="close this track"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <span
                  class="MuiTypography-root makeStyles-trackName MuiTypography-body1"
                >
                  Reference Sequence (volvox)
                </span>
                <button
                  aria-controls="simple-menu"
                  aria-haspopup="true"
                  class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                  data-testid="track_menu_icon"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiPaper-root makeStyles-trackRenderingContainer MuiPaper-outlined MuiPaper-rounded"
                data-testid="trackRenderingContainer-test_view-volvox_refseq"
                role="presentation"
                style="height: 180px;"
              >
                <div
                  class="makeStyles-renderingComponentContainer"
                  style="transform: scaleX(1);"
                >
                  <div
                    class="makeStyles-display"
                    data-testid="display-volvox_refseq-LinearReferenceSequenceDisplay"
                    role="presentation"
                  >
                    <div
                      class="makeStyles-linearBlocks"
                      data-testid="Blockset"
                      style="left: -1024px;"
                    >
                      <div
                        class="makeStyles-boundaryPaddingBlock"
                        style="background: none; width: 1024px;"
                      />
                      <div
                        class="makeStyles-contentBlock"
                        style="width: 1040px;"
                      >
                        <div
                          style="display: flex;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-overlay"
                style="height: 180px;"
              />
              <div
                class="makeStyles-horizontalHandle makeStyles-resizeHandle"
                data-resizer="true"
                role="presentation"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
